Doc test failure

### Description

When running the unit tests locally I get the following failed test:
```
===================================================================================================== FAILURES ======================================================================================================
_________________________________________________________________________________________________ [doctest] io.rst __________________________________________________________________________________________________
104 
105     >>> from astropy import units as u
106     >>> from astropy.timeseries import BinnedTimeSeries
107     >>> binned_filename = get_pkg_data_filename('data/binned.csv',
108     ...                                          package='astropy.timeseries.tests')
109     >>> ts = BinnedTimeSeries.read(binned_filename, format='ascii.csv',
110     ...                            time_bin_start_column='time_start',
111     ...                            time_bin_size_column='bin_size',
112     ...                            time_bin_size_unit=u.s)
113     >>> ts[:3]
Differences (unified diff with -expected +actual):
    @@ -1,8 +1,8 @@
     <BinnedTimeSeries length=3>
    -     time_bin_start     time_bin_size ...    E       F
    -                              s       ...
    -          Time             float64    ... float64 float64
    ------------------------ ------------- ... ------- -------
    -2016-03-22T12:30:31.000           3.0 ...   28.87   63.44
    -2016-03-22T12:30:34.000           3.0 ...   27.76   59.98
    -2016-03-22T12:30:37.000           3.0 ...   27.04   59.61
    +     time_bin_start     time_bin_size         time_end           A       B       C       D       E       F   
    +                              s                                                                              
    +          Time             float64             str23          float64 float64 float64 float64 float64 float64
    +----------------------- ------------- ----------------------- ------- ------- ------- ------- ------- -------
    +2016-03-22T12:30:31.000           3.0 2016-03-22T12:30:34.000  164.93  114.73   26.27   19.21   28.87   63.44
    +2016-03-22T12:30:34.000           3.0 2016-03-22T12:30:37.000  164.89  114.75   26.22   19.07   27.76   59.98
    +2016-03-22T12:30:37.000           3.0 2016-03-22T12:30:40.000  164.63  115.04   25.78   19.01   27.04   59.61

/Users/wjamieson/Workspaces/astropy/astropy/docs/timeseries/io.rst:113: DocTestFailure
`

### Expected behavior

This should not fail.

### How to Reproduce

1. Get package from '...'
2. Then run '...'
3. An error occurs.

```python
# Put your Python code snippet here.
```
```
pip install -e ".[all,test_all,docs]"`
pytest
```

It fails on tox too.

### Versions

```python
import astropy
try:
    astropy.system_info()
except AttributeError:
    import platform; print(platform.platform())
    import sys; print("Python", sys.version)
    import astropy; print("astropy", astropy.__version__)
    import numpy; print("Numpy", numpy.__version__)
    import erfa; print("pyerfa", erfa.__version__)
    try:
        import scipy
        print("Scipy", scipy.__version__)
    except ImportError:
        print("Scipy not installed")
    try:
        import matplotlib
        print("Matplotlib", matplotlib.__version__)
    except ImportError:
        print("Matplotlib not installed")
```
```
Using Python 3.12.7 environment at /Users/wjamieson/.pyenv/versions/3.12.7/envs/astropy
Package                       Version                   Editable project location
----------------------------- ------------------------- -------------------------------------------
accessible-pygments           0.0.5
aiobotocore                   2.15.2
aiohappyeyeballs              2.4.3
aiohttp                       3.10.10
aioitertools                  0.12.0
aiosignal                     1.3.1
alabaster                     1.0.0
appnope                       0.1.4
array-api-strict              2.1
asdf                          3.5.0
asdf-astropy                  0.6.1
asdf-coordinates-schemas      0.3.0
asdf-standard                 1.1.1
asdf-transform-schemas        0.5.0
astropy                       7.0.0.dev1261+g68a330049c /Users/wjamieson/Workspaces/astropy/astropy
astropy-iers-data             0.2024.10.21.0.33.21
astropy-sphinx-theme          1.1
asttokens                     2.4.1
attrs                         24.2.0
babel                         2.16.0
beautifulsoup4                4.12.3
bleach                        6.1.0
botocore                      1.35.36
bottleneck                    1.4.2
cachetools                    5.5.0
certifi                       2024.8.30
cfgv                          3.4.0
chardet                       5.2.0
charset-normalizer            3.4.0
click                         8.1.7
cloudpickle                   3.1.0
colorama                      0.4.6
comm                          0.2.2
contourpy                     1.3.0
coverage                      7.6.4
cycler                        0.12.1
dask                          2024.10.0
debugpy                       1.8.7
decorator                     5.1.1
distlib                       0.3.9
docutils                      0.21.2
execnet                       2.1.1
executing                     2.1.0
filelock                      3.16.1
fonttools                     4.54.1
frozenlist                    1.4.1
fsspec                        2024.10.0
h5py                          3.12.1
html5lib                      1.1
hypothesis                    6.115.3
identify                      2.6.1
idna                          3.10
imagesize                     1.4.1
iniconfig                     2.0.0
ipykernel                     6.29.5
ipython                       8.28.0
ipywidgets                    8.1.5
jedi                          0.19.1
jinja2                        3.1.4
jmespath                      1.0.1
jplephem                      2.22
jupyter-client                8.6.3
jupyter-core                  5.7.2
jupyterlab-widgets            3.0.13
kiwisolver                    1.4.7
locket                        1.0.0
markupsafe                    3.0.2
matplotlib                    3.9.2
matplotlib-inline             0.1.7
mpmath                        1.3.0
multidict                     6.1.0
nest-asyncio                  1.6.0
nodeenv                       1.9.1
numpy                         2.1.2
numpydoc                      1.8.0
objgraph                      3.6.2
packaging                     24.1
pandas                        2.2.3
pandas-stubs                  2.2.3.241009
parso                         0.8.4
partd                         1.4.2
pexpect                       4.9.0
pillow                        11.0.0
pip                           24.2
platformdirs                  4.3.6
pluggy                        1.5.0
pre-commit                    4.0.1
prompt-toolkit                3.0.48
propcache                     0.2.0
psutil                        6.1.0
ptyprocess                    0.7.0
pure-eval                     0.2.3
pyarrow                       17.0.0
pydata-sphinx-theme           0.15.4
pyerfa                        2.0.1.4
pygments                      2.18.0
pyparsing                     3.2.0
pyproject-api                 1.8.0
pytest                        8.3.3
pytest-arraydiff              0.6.1
pytest-astropy                0.11.0
pytest-astropy-header         0.2.2
pytest-cov                    5.0.0
pytest-doctestplus            1.2.1
pytest-filter-subpackage      0.2.0
pytest-mock                   3.14.0
pytest-remotedata             0.4.1
pytest-xdist                  3.6.1
python-dateutil               2.9.0.post0
pytz                          2024.2
pyyaml                        6.0.2
pyzmq                         26.2.0
requests                      2.32.3
s3fs                          2024.10.0
scipy                         1.14.1
semantic-version              2.10.0
setuptools                    75.2.0
sgp4                          2.23
six                           1.16.0
skyfield                      1.49
snowballstemmer               2.2.0
sortedcontainers              2.4.0
soupsieve                     2.6
sphinx                        8.1.3
sphinx-astropy                1.9.1
sphinx-automodapi             0.18.0
sphinx-changelog              1.6.0
sphinx-copybutton             0.5.2
sphinx-design                 0.6.1
sphinx-gallery                0.18.0
sphinxcontrib-applehelp       2.0.0
sphinxcontrib-devhelp         2.0.0
sphinxcontrib-globalsubs      0.1.2
sphinxcontrib-htmlhelp        2.1.0
sphinxcontrib-jquery          4.1
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          2.0.0
sphinxcontrib-serializinghtml 2.0.0
stack-data                    0.6.3
tabulate                      0.9.0
threadpoolctl                 3.5.0
toolz                         1.0.0
tornado                       6.4.1
towncrier                     24.8.0
tox                           4.23.0
traitlets                     5.14.3
types-pytz                    2024.2.0.20241003
typing-extensions             4.12.2
tzdata                        2024.2
urllib3                       2.2.3
virtualenv                    20.27.0
wcwidth                       0.2.13
webencodings                  0.5.1
widgetsnbextension            4.0.13
wrapt                         1.16.0
yarl                          1.15.5
```

--- a/conftest.py
+++ b/conftest.py
@@ -18,6 +18,13 @@ except ImportError:
 
 # This has to be in the root dir or it will not display in CI.
 def pytest_configure(config):
+    # Strip COLUMNS/LINES so shutil.get_terminal_size() falls back to (80, 24)
+    # in the test session. Interactive shells (bash, zsh) export these on
+    # window resize, and pytest inherits them, so otherwise tests and doctests
+    # that read the terminal size are non-deterministic across environments.
+    os.environ.pop("COLUMNS", None)
+    os.environ.pop("LINES", None)
+
     PYTEST_HEADER_MODULES["PyERFA"] = "erfa"
     PYTEST_HEADER_MODULES["Cython"] = "cython"
     PYTEST_HEADER_MODULES["Scikit-image"] = "skimage"

